doc/source/reference/c-api/types-and-structures.rst

```diff
@@ -71,23 +71,30 @@ PyArray_Type and PyArrayObject
    member of this structure contains a pointer to the :c:data:`PyArray_Type`
    typeobject.
 
+.. c:type:: PyArrayObject_fields
+
+   An implementation detail for accessing the :c:data:`PyArrayObject` struct.
+   You should use :c:data:`PyArrayObject` to ensure code compatibility and stability.
+
```

Comment:
I don't think we should document this internal struct in the documentation of public structures. We hid it on purpose. We could have internal documentation that could document this and other structs.